cachix/src/Cachix/Client/OptionsParser.hs

```diff
@@ -127,11 +120,13 @@ parserCachixCommand =
                 )
     use =
       Use <$> nameArg
-        <*> ( UseOptions
-                <$> switch
-                      ( long "nixos"
-                          <> short 'n'
-                          <> help "Write NixOS modules"
+        <*> ( InstallationMode.UseOptions
+                <$> optional
+                      ( option (maybeReader InstallationMode.fromString)
+                          ( long "mode"
+                              <> short 'm'
+                              <> help "Mode in which to configure binary caches for Nix. Supported values: nixos, root-nixconf, user-nixconf"
```

Comment:
This list should be generated using `Bounded` and `Enum` instances.

Reply:
I wanted to use that, but there were two issues:

- can't derive since it has more than one constructor
- we don't want to show `UntrustedRequiresSudo` as an option since that's an error

So I'll just add a comment to ADT to update this list when adding/removing stuff

Reply:
:+1: 
I guess you could add a type with Enum for just this purpose, but that may not be a real improvement.